ci/travis/dump-reports.sh

```diff
@@ -0,0 +1,49 @@
+#!/usr/bin/env bash
+#
+# Copyright 2018 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+set -eu
+
+if [[ -z "${PROJECT_ROOT+x}" ]]; then
+  readonly PROJECT_ROOT="$(cd "$(dirname $0)/../.."; pwd)"
+fi
+source "${PROJECT_ROOT}/ci/travis/linux-config.sh"
+
+# Find any analysis reports, currently ABI checks and Clang static analysis are
+# the two things that produce them. Note that the Clang static analysis reports
+# are copied into the scan-build-output directory by the build-docker.sh script.
+readonly ABI_CHECK_REPORTS="$(find "${BUILD_OUTPUT}" -name 'compat_report.html')"
+readonly SCAN_BUILD_REPORTS="$(find scan-build-output/ -name '*.html' 2>/dev/null)"
+
+if [ -z "${ABI_CHECK_REPORTS}" ] && [ -z "${SCAN_BUILD_REPORTS}" ]; then
+  echo "No analysis reports found, exit scripts with success."
+  exit 0
+fi
+
+# If w3m is installed there is nothing to do.
+if which w3m >/dev/null 2>&1; then
+  echo "Found w3m already installed."
+else
+  # Try to install a HTML renderer, if this fails the script will exit.
+  # Note that this runs on the Travis VM, under Ubuntu, so the command
+  # to install things is well-known:
+  sudo apt install -y w3m
+fi
+
+for report in ${ABI_CHECK_REPORTS} ${SCAN_BUILD_REPORTS}; do
```

Comment:
I concur, given that we do quote "${report}" itself.

Reply:
Quotes would not work because these are really lists of files: `${ABI_CHECK_REPORTS}` expands to something like `/v/foo/bar.html /v/foo/baz.html` . On the other hand they would not work if the files themselves have spaces in them, but that is a problem for another day.

Reply:
Happy to fix it if somebody has a suggestion on how to do so btw.

Reply:
In that case, we should really be putting these things into an array and iterating over the array, like in this admittedly gross answer:
https://stackoverflow.com/questions/8213328/store-the-output-of-find-command-in-an-array/8213509

Reply:
```
mkdir /tmp/test
touch '/tmp/test/file.ext'
touch '/tmp/test/another file.ext'
touch '/tmp/test/yet another file.ext'

readonly IFS_SAV="${IFS}"
IFS=$'\n' LIST=( $(find /tmp/test -name *.ext) )
IFS="${IFS_SAV}"

for ELEM in "${LIST[@]}"; do
  echo "ELEM=${ELEM}"
done
```

Reply:
And perhaps you don't even need that `${IFS_SAV}` stuff.

Reply:
Aside: `IFS=$'\0' LIST=( $(find /tmp/test -name *.ext -print0) )` doesn't look like it works unfortunately (to allow file names with newlines).